client/components/social-buttons/google.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import SocialLogo from 'social-logos';
+
+export default class GoogleLoginButton extends React.Component{
+	constructor( props ) {
+		super( props );
+		this.state = {
+			auth2: null,
+		};
+		this.handleClick = this.handleClick.bind( this );
+	}
+
+	componentDidMount () {
+		this.loadDependency().then( gapi => {
+			gapi.load( 'auth2', () => {
+				const auth2 = gapi.auth2.init( {
+					client_id: this.props.clientId,
+					fetch_basic_profile: this.props.fetchProfile,
+					scope: this.props.scope
+				} );
+				this.setState( { auth2 } );
+			} );
+		} )
+	}
+
+	loadDependency() {
+		return new Promise( resolve => {
+			if ( window.gapi ) {
+				resolve( window.gapi );
+			} else {
+				( function( d, s, id ) {
+					var js, gs = d.getElementsByTagName( s )[0];
+					if ( d.getElementById( id ) ) { return; }
+					js = d.createElement( s ); js.id = id;
+					js.src = 'https://apis.google.com/js/platform.js';
+					gs.parentNode.insertBefore( js, gs );
+				}( document, 'script', 'google-platform' ) );
+
+				const scriptTag = document.getElementById( 'google-platform' );
+				scriptTag.onload = () => resolve( window.gapi );
+			}
+		} );
+	}
+
+	handleClick() {
+		const { responseHandler } = this.props;
+
+		this.state.auth2.signIn().then( googleUser => {
+			responseHandler( googleUser );
+		} );
+	}
+
+	render() {
+		// Show an empty button if Google auth2 API is not loaded
+		if ( ! this.state.auth2 ) {
+			return (
+				<button className="button">&nbsp;</button>
```

Comment:
I think we probably want to tweak the loading state here, as currently a single blank button is displayed (when both are loading):

![screen shot 2017-03-16 at 12 36 35 pm](https://cloud.githubusercontent.com/assets/1130674/24015325/6875c9de-0a45-11e7-8943-39e6ab53ff9e.png)

I think, ideally, we would allow users to click this button early and simply call `auth2.signIn()` once the script finished loading, but I think just improving the loading state so that it says `Loading…` next to the logo would be sufficient for now.